internal/applier/helm_test.go

```diff
@@ -0,0 +1,315 @@
+		mag.dryRunInstallOK = false
+		return mag.desiredRel, nil
+	}
+	return mag.desiredRel, mag.installErr
+}
+
+func (mag *mockActionGetter) Upgrade(name, namespace string, chrt *chart.Chart, vals map[string]interface{}, opts ...helmclient.UpgradeOption) (*release.Release, error) {
+	if mag.dryRunUpgradeOK {
+		mag.dryRunUpgradeOK = false
+		return mag.desiredRel, nil
+	}
+	return mag.desiredRel, mag.upgradeErr
+}
+
+func (mag *mockActionGetter) Uninstall(name string, opts ...helmclient.UninstallOption) (*release.UninstallReleaseResponse, error) {
+	return nil, nil
+}
+
+func (mag *mockActionGetter) Reconcile(rel *release.Release) error {
+	return mag.reconcileErr
+}
+
+var (
+	// required for unmockable call to convert.RegistryV1ToHelmChart
+	validFS = fstest.MapFS{
+		"metadata/annotations.yaml": &fstest.MapFile{Data: []byte("{}")},
+		"manifests": &fstest.MapFile{Data: []byte(`apiVersion: operators.operatorframework.io/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: test.v1.0.0
+  annotations:
+    olm.properties: '[{"type":"from-csv-annotations-key", "value":"from-csv-annotations-value"}]'
+spec:
+  installModes:
+    - type: AllNamespaces
+      supported: true`)},
+	}
+
+	// required for unmockable call to util.ManifestObjects
+	validManifest = `apiVersion: v1
+kind: Service
+metadata:
+  name: service-a
+  namespace: ns-a
+spec:
+  clusterIP: None
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: service-b
+  namespace: ns-b
+spec:
+  clusterIP: 0.0.0.0`
+
+	testCE            = &v1.ClusterExtension{}
+	testObjectLabels  = map[string]string{"object": "label"}
+	testStorageLabels = map[string]string{"storage": "label"}
+)
+
+func TestApply_Base(t *testing.T) {
+	t.Run("fails converting content FS to helm chart", func(t *testing.T) {
+		helmApplier := applier.Helm{}
+
+		objs, state, err := helmApplier.Apply(context.TODO(), os.DirFS("/"), testCE, testObjectLabels, testStorageLabels)
+		require.Error(t, err)
+		require.Nil(t, objs)
+		require.Empty(t, state)
+	})
+
+	t.Run("fails trying to obtain an action client", func(t *testing.T) {
+		mockAcg := &mockActionGetter{actionClientForErr: errors.New("failed getting action client")}
+		helmApplier := applier.Helm{ActionClientGetter: mockAcg}
+
+		objs, state, err := helmApplier.Apply(context.TODO(), validFS, testCE, testObjectLabels, testStorageLabels)
+		require.Error(t, err)
+		require.ErrorContains(t, err, "getting action client")
+		require.Nil(t, objs)
+		require.Empty(t, state)
+	})
+
+	t.Run("fails getting current release and !driver.ErrReleaseNotFound", func(t *testing.T) {
+		mockAcg := &mockActionGetter{getClientErr: errors.New("failed getting current release")}
+		helmApplier := applier.Helm{ActionClientGetter: mockAcg}
+
+		objs, state, err := helmApplier.Apply(context.TODO(), validFS, testCE, testObjectLabels, testStorageLabels)
+		require.Error(t, err)
+		require.ErrorContains(t, err, "getting current release")
+		require.Nil(t, objs)
+		require.Empty(t, state)
```

Comment:
Should not, in this case, we validate that the state is the expected one?

Reply:
it is being validated. The expected valid state here and in other places like that is an empty string